### PR TITLE
add port field to iperf

### LINKED ISSUE #2082: os-iperf: Add option to set port
**Important notices**
Before you add a new report, we ask you kindly to acknowledge the following:
[x] I have read the contributing guide lines at https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.md
[x] I have searched the existing issues and I'm convinced that mine is new.
[x] When the request is meant for an existing plugin, I've added its name to the title.

It would be nice to have an option to set port for Iperf because setting port on clients on every run is very annoying.

**Describe the solution you'd like**
I would like an input field in the `/ui/iperf`.

### PR DESCRIPTION
**Pull request by Marco98**
Fixes #2082 

My first PR to opnsense. Hope i did everything right :smile:

### CODE REVIEW

File: benchmarks/iperf/src/opnsense/scripts/iperf/ruby_iperf.rb [hunk 1]
```diff
@@ -182,15 +186,20 @@ def run_test_thread(interface = 'any')
...
+            port = 0
+            if command.length > 1
+              intf = command[1]
               # check if a valid interface was given
               interface = intf if intf =~ /^[a-z0-9_-]+$/
             end
-            data = run_test_thread interface
+            if command.length > 2
+              port = command[2].to_i
+              port = 0 if !find_open_ports.include? port
```
Comment (fabianfrz) [inline benchmarks/iperf/src/opnsense/scripts/iperf/ruby_iperf.rb:200]: Ruby: `unless condition` is equal to `if !(condition)`. It makes it easier to read.

```suggestion
              port = 0 unless find_open_ports.include? port
```

**Review by fabianfrz — CHANGES_REQUESTED** [review]
Some little things.

File: benchmarks/iperf/src/opnsense/mvc/app/controllers/OPNsense/iperf/Api/InstanceController.php [hunk 2]
```diff
@@ -54,10 +55,15 @@ public function setAction()
             return array('status' => 'error',
                     'error' => 'interface parameter is missing');
         }
+        if (!isset($_POST['instance']['listenPort'])) {
+            return array('status' => 'error',
+                    'error' => 'listenPort parameter is missing');
+        }
         $interface_name = $_POST['instance']['interface'];
+        $listen_port = $_POST['instance']['listenPort'];
```
[the remark this answers is not in the document]
Comment (Marco98) [reply benchmarks/iperf/src/opnsense/mvc/app/controllers/OPNsense/iperf/Api/InstanceController.php]: both implemented now

File: benchmarks/iperf/src/opnsense/mvc/app/models/OPNsense/iperf/FakeInstance.xml [hunk 3]
```diff
@@ -1,11 +1,16 @@
 <model>
   <mount>//OPNsense/Iperf3</mount>
   <description>Fake model for the API - will be never stored to config (only used for defaults, validation etc.).</description>
+  <version>1.0.0</version>
```
[the remark this answers is not in the document]
Comment (Marco98) [reply benchmarks/iperf/src/opnsense/mvc/app/models/OPNsense/iperf/FakeInstance.xml]: removed version now

File: benchmarks/iperf/src/opnsense/scripts/iperf/ruby_iperf.rb (hunk 1 above)
Comment (Marco98) [reply benchmarks/iperf/src/opnsense/scripts/iperf/ruby_iperf.rb]: changed now as recommended

**Comment by AdSchellevis** [comment]
timeout

**Pull request closed without merging**